lib/ansible/utils/hashing.py

```diff
@@ -48,14 +48,13 @@ def secure_hash(filename, hash_func=sha1):
     digest = hash_func()
     blocksize = 64 * 1024
     try:
-        infile = open(to_bytes(filename, errors='surrogate_or_strict'), 'rb')
-        block = infile.read(blocksize)
-        while block:
-            digest.update(block)
+        with open(to_bytes(filename, errors='surrogate_or_strict'), 'rb') as infile:
```

Comment:
Controller code no longer needs to convert paths to bytes:

```suggestion
        with open(filename, 'rb') as infile:
```